plugins/modem-manager/fu-mm-device.c

```diff
@@ -861,6 +1024,23 @@ fu_mm_device_udev_new (MMManager *manager,
 	fu_device_set_name (FU_DEVICE (self), info->name);
 	fu_device_set_version (FU_DEVICE (self), info->version);
 	self->update_methods = info->update_methods;
+	if (self->update_methods & MM_MODEM_FIRMWARE_UPDATE_METHOD_MBIM_QDU) {
```

Comment:
Is all this new block really needed? Why? At this point we're already in the upgrade process for the device and we can create the udev device with the info coming from the old MM device. If we needed to have vendor ids and protocol added, we could have added those items to the FuPluginMmInhibitedDeviceInfo but I don't really think you need them.

Reply:
Also, following another comment in this review, the changes here are exclusively required when the layout of the module in the kernel changes (e.g. from modem-mode to download-mode, using a totally different layout). In your case, the module was MBIM in modem-mode and MBIM in download-mode, so there should be no need to handle it as a udev MM device.